Sources/Action.swift

```diff
@@ -109,21 +95,21 @@ public final class Action<Input, Output, Error: Swift.Error> {
 	///            producer.
 	public func apply(_ input: Input) -> SignalProducer<Output, ActionError<Error>> {
 		return SignalProducer { observer, disposable in
-			var startedExecuting = false
+			var executedProducer: SignalProducer<Output, Error>?
 
-			self.executingQueue.sync {
-				if self._isEnabled.value {
-					self._isExecuting.value = true
-					startedExecuting = true
+			self._state.modify { state in
+				if state.isEnabled {
+					state.isExecuting = true
+					executedProducer = self.executeClosure(input)
```

Comment:
You can `return` this from the `modify` directly instead of assigning to a `var`.